src/homeblks/volume/volume.cpp

```diff
@@ -334,8 +334,119 @@ indx_tbl* Volume::recover_indx_tbl(btree_super_block& sb, btree_cp_sb& cp_info)
                            SnapMgr::add_read_tracker, &cp_info);
     return static_cast< indx_tbl* >(tbl);
 }
+static std::vector< bool > find_non_zero_data(const uint8_t* buf, size_t size, uint32_t nlbas) {
+    std::vector< bool > empty_blocks;
+
+    auto is_buf_empty = [](const uint8_t* buf, size_t size) -> bool {
+        return buf[0] == 0 && !std::memcmp(buf, buf + 1, size - 1);
```

Comment:
as we discussed, you are going to change it with the intel crc algorithm to detect zero then do a bitwise compare, right?

Reply:
affirmative. I will make this change and will update this pr